pages/_app.tsx

```diff
@@ -85,7 +102,7 @@ const App: FC<{
         )}
       </Head>
       <SafeHydrate>
-        <AppProvider relayCache={relayCache}>
+        <AppProvider relayEnvironment={relayEnvironment}>
```

Comment:
we have to pass in a relay environment since we make the requests earlier than the `AppProvider` is created.